Simplify the structure of the introduction (#80)
xsleaks/wiki: content/_index.md

# XS-Leaks Wiki
## Overview

> Cross-site leaks (aka XS-Leaks, XSLeaks) are a class of vulnerabilities using side-channels [^side-channel] built into the web platform. They take advantage of the web's core principle of composability, which allows websites to interact with each other, and abuse legitimate mechanisms to infer information about the user.

Reviewer: To me, XS-Leaks are a subset of side-channel techniques, so saying that they "use side-channel techniques" sounds wrong to me.  I like the word `derived` to be honest. 

A note, this also makes for a definition of XS-Leak rather than the overview per se.

Author: I guess here I was considering "side-channels" as behaviors that leak information, rather than as techniques to exploit these behaviors. A clearer way to capture the meaning I was going for could be: "Cross-site leaks (aka XS-Leaks, XSLeaks) are a class of vulnerabilities which abuse side-channels [^side-channel] built into the web platform."

(My slight concern about the original wording is that vulnerabilities themselves seem somewhat distinct from techniques used to exploit them. To me an XS-leak is a behavior that allows information to be leaked by an attacker, rather than the method the attacker uses to learn this information. But maybe this is not really how XS-Leaks folks interpret this term?)

If the updated wording above sounds off, do you have an idea for tweaking the original text so that it more clearly states whether an XS-leaks are vulnerabilities or exploits? E.g. in the latter case we could say "Cross-Site Leaks (aka XS-Leaks, XSLeaks) are exploits derived from browser side-channel techniques which abuse legitimate web browser behaviors" -- though this seems slightly less clear to me.

Reviewer: > To me an XS-leak is a behavior that allows information to be leaked by an attacker, rather than the method the attacker uses to learn this information

Exactly, that's why I don't think that `using` or `abuse` because it implies that XS-Leak is an attack, but it's not how I perceive them. But I now see what your point is, and we should rather change `side-channel techniques` to something else, because we want to retrieve the principle from side-channel techniques and define them as XS-Leaks!

Reviewer: Added comparison to CSRF at the end of the definition, suggested by @goedi02 in the doc. I still don't think `using side-channel` is an adequate way to describe XS-Leaks. Derived sounds better to me, but I agree that this still could be further improved to get a better sight on them. 

```suggestion
Cross-site leaks (aka XS-Leaks, XSLeaks) are a class of vulnerabilities derived from side-channels [^side-channel] built into the web platform. They take advantage of the web's core principle of composability, which allows websites to interact with each other, and abuse legitimate mechanisms to infer information about the user. One way of looking at XS-Leaks is to highlight their similarity with cross-site request forgery (CSRF) techniques, with the main difference being that instead of allowing other websites to perform actions on behalf of a user, XS-Leaks can be used to infer information about a user. 
```

@arturjanc what do you think?

> Browsers provide a wide variety of features to support interactions between different web applications; for example, they permit a website to load subresources, navigate, or send messages to another application. While such behaviors are generally constrained by security mechanisms built into the web platform (e.g. the [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)), XS-Leaks take advantage of small pieces of information which are exposed during interactions between websites. This information may let an attacker reveal sensitive information about users, such as their data in other web applications, details about their local environment, or internal networks they are connected to.

Reviewer: If we change the next heading, then maybe it would be worth mentioning that this is a principle of an xs-leaks somewhere in the paragraph?

Author: Done.

> ## The principle of an XS-Leak

Reviewer: After we moved the text to the overview, The principle of an XS-Leak is not clear to me from the text that is left. 

```suggestion
## Cross-site oracles
```

Maybe changing the heading to something like that?

Author: Done.

# XS-Leaks Wiki
## Overview

> Cross-site leaks (aka XS-Leaks, XSLeaks) are a class of vulnerabilities derived from side-channels [^side-channel] built into the web platform. They take advantage of the web's core principle of composability, which allows websites to interact with each other, and abuse legitimate mechanisms to infer information about the user. One way of looking at XS-Leaks is to highlight their similarity with cross-site request forgery (CSRF [^csrf]) techniques, with the main difference being that instead of allowing other websites to perform actions on behalf of a user, XS-Leaks can be used to infer information about a user. 

Reviewer: Restoring browser-features footnote into the definition, we can drop it later on. 

```suggestion
Cross-site leaks (aka XS-Leaks, XSLeaks) are a class of vulnerabilities derived from side-channels [^side-channel] built into the web platform. They take advantage of the web's core principle of composability, which allows websites to interact with each other, and abuse legitimate mechanisms [^browser-features] to infer information about the user. One way of looking at XS-Leaks is to highlight their similarity with cross-site request forgery (CSRF [^csrf]) techniques, with the main difference being that instead of allowing other websites to perform actions on behalf of a user, XS-Leaks can be used to infer information about a user. 
```

Browsers provide a wide variety of features to support interactions between different web applications; for example, they permit a website to load subresources, navigate, or send messages to another application. While such behaviors are generally constrained by security mechanisms built into the web platform (e.g. the [same-origin policy](https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy)), XS-Leaks take advantage of small pieces of information which are exposed during interactions between websites. 

> The principle of an XS-Leak is to use such side-channels available on the web to reveal sensitive information about users, such as their data in other web applications, details about their local environment, or internal networks they are connected to.

Author: @terjanq I moved the "principle of an XS-leak" mention directly into text and updated the following section heading as you suggested, and I think it reads better.

WDYT?

Reviewer: LGTM

## References
[^side-channel]: Side Channel Vulnerabilities on the Web - Detection and Prevention, [link](https://owasp.org/www-pdf-archive/Side_Channel_Vulnerabilities.pdf)
[^csrf]: Cross Site Request Forgery (CSRF), [link](https://owasp.org/www-community/attacks/csrf)

> [^browser-features]: In some cases, these features are maintained to preserve backwards compatibility. But, in other cases, new features are added to browsers regardless of the fact that they introduce potential cross-site leaks (e.g. [Scroll to Text Fragment]({{< ref "scroll-to-text-fragment.md" >}})), as the benefits are considered to outweigh the downsides.

Reviewer: Looks like [^browser-features] footnote is not used in the definition anymore, does it make sense to restore it?

Reviewer: In: https://github.com/xsleaks/wiki/pull/80/commits/d2ad0083f3626f3a4040a503c0c4d0a922ecbe37

## References
[^side-channel]: Side Channel Vulnerabilities on the Web - Detection and Prevention, [link](https://owasp.org/www-pdf-archive/Side_Channel_Vulnerabilities.pdf)

> [^csrf]: Cross Site Request Forgery (CSRF), [link](https://owasp.org/www-community/attacks/csrf)

Reviewer: Don't remove the line if we leave CSRF in the definition.
